go/build.gradle

```diff
@@ -0,0 +1,81 @@
+import io.deephaven.tools.docker.DiffTask
+
+plugins {
+    id 'com.bmuschko.docker-remote-api'
+    id 'io.deephaven.project.register'
+    id 'io.deephaven.deephaven-in-docker'
+}
+
+evaluationDependsOn Docker.registryProject('go')
+
+
+// The Go convention is to check in generated sources - while there is a "central repo server" for Go, it is a proxy
+// that merely polls each proxy several times an hour, but doesn't allow contents to be deployed to it. As such,
+// any generated code must exist in git, and will be downloaded from there. As part of the build then, we generate and
+// compare/overwrite protobuf sources.
+configurations {
+    go {}
+}
+dependencies {
+    go project(path: ':proto:proto-backplane-grpc', configuration: 'go')
+}
+def compare = tasks.register('compareProtobuf', DiffTask) {
+    expectedContents.set configurations.go
+    actualContents.set layout.projectDirectory
+    generateTask.set ':go:updateProtobuf'
+
+    ignore {
+        exclude 'internal/proto/**'
+    }
+}
+// fail a "check" build if these are out of date
+tasks.getByName('quick').dependsOn(compare)
+
+def update = tasks.register('updateProtobuf', Sync) {
+    finalizedBy compare
+    from configurations.go
+    into layout.projectDirectory
+    preserve {
+        exclude 'internal/proto/**'
+    }
+}
+
+
+// start a grpc-api server
+String randomSuffix = UUID.randomUUID().toString();
+deephavenDocker {
+    containerName.set "dh-server-for-go-${randomSuffix}"
+    networkName.set "go-test-network-${randomSuffix}"
+}
+def testGoClient = Docker.registerDockerTask(project, 'testGoClient') {
```

Comment:
I've tried to get this task marked as `shouldRunAfter compare` - I think we should arguably do the compare as early on as possible, but it didn't seem to work. We may really mean the *first* or *all* tasks of `registerDockerTask`. Instead of the returned one. Not a blocker to merge, but may be a nicety we can do if easy.

Maybe there is really something more general we should add to our setup: `quick` tasks should preferably be run before other check/test tasks.